pkg/interface/link/src/js/components/lib/link-detail-preview.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react';
+import { Route, Link } from "react-router-dom";
+import { base64urlEncode } from "../../lib/util";
+import moment from "moment";
+
+export class LinkPreview extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      timeSinceLinkPost: this.getTimeSinceLinkPost(this.props.data)
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props) {
+      if (this.state.timeSinceLinkPost === "") {
+        this.setState({
+          timeSinceLinkPost: this.getTimeSinceLinkPost(this.props.data)
+        });
+      }
+    }
+  }
+
+  componentDidMount() {
+    this.updateTimeSinceNewestMessageInterval = setInterval(() => {
+      this.setState({
+        timeSinceLinkPost: this.getTimeSinceLinkPost(this.props.data)
+      });
+    }, 60000);
+  }
+
+  componentWillUnmount() {
+    if (this.updateTimeSinceNewestMessageInterval) {
+      clearInterval(this.updateTimeSinceNewestMessageInterval);
+      this.updateTimeSinceNewestMessageInterval = null;
+    }
+  }
+
+  getTimeSinceLinkPost(data) {
+    return !!data.time
+      ? moment.unix(data.time / 1000).from(moment.utc())
+      : "";
+  }
+
+  render() {
+    const { props } = this;
+
+    let URLparser = new RegExp(
+      /((?:([\w\d\.-]+)\:\/\/?){1}(?:(www)\.?){0,1}(((?:[\w\d-]+\.)*)([\w\d-]+\.[\w\d]+))){1}(?:\:(\d+)){0,1}((\/(?:(?:[^\/\s\?]+\/)*))(?:([^\?\/\s#]+?(?:.[^\?\s]+){0,1}){0,1}(?:\?([^\s#]+)){0,1})){0,1}(?:#([^#\s]+)){0,1}/
+    );
+
+    let hostname = URLparser.exec(props.url);
+
+    if (hostname) {
+      hostname = hostname[4];
+    }
+
+    let imgMatch = /(jpg|img|png|gif|tiff|jpeg|JPG|IMG|PNG|TIFF|GIF|webp|WEBP|webm|WEBM)$/.exec(
+      props.url
+    );
+
+    let youTubeRegex = new RegExp(
+      "" +
+      /(?:https?:\/\/(?:[a-z]+.)?)/.source + // protocol
+      /(?:youtu\.?be(?:\.com)?\/)(?:embed\/)?/.source + // short and long-links
+        /(?:(?:(?:(?:watch\?)?(?:time_continue=(?:[0-9]+))?.+v=)?([a-zA-Z0-9_-]+))(?:\?t\=(?:[0-9a-zA-Z]+))?)/.source // id
+    );
+
+    let ytMatch = youTubeRegex.exec(props.url);
+
+    let embed = "";
+
+    if (imgMatch) {
+      embed = <img className="db w-50"
```

Comment:
> Tiny images will get stretched up to around ~300px. I can't find a way to cap the maximum image size to its native size.

```suggestion
      embed = <img className="db"
```

(^:

Reply:
(Suggestion: might be nice for the image embeds to be clickable links just like the title, so you can "click to expand".)